### ISSUE TITLE
access `regex::Match`  or `Captures` struct impl results (e.g. range, start, end, etc)

### ISSUE THREAD

**Issue by rtbs-dev** [apache/arrow-rs#5966]
**What are you trying to do?**
I would like to extract `regex::Captures` structs directly, rather than the already-unwrapped string values, because I require the byte offsets directly (e.g. to implement [ISO 24612](https://www.iso.org/standard/37326.html), which requires primacy of string span locations in a document, not the contents themselves). 

**Describe the solution you'd like**

Either a new function to extract the [`Captures`](https://docs.rs/regex/latest/regex/struct.Captures.html) structs directly, or a mode for `compute::regexp_match` that provides the offset anchors for each match.  

**Describe alternatives you've considered**

Retrieving the strings and then trying to find their locations one-by-one is wasteful of resources, and I can't find a flag to enable the desired behavior :) 

**Additional context**

Coming here from [polars#16341](https://github.com/pola-rs/polars/issues/16341), but if I'm understanding their codebase correctly, they are using this backend as an intermediary to the `regex` rust lib.

**Comment by tustvold** [comment]
I don't believe polars is making use of arrow-rs, in favour of its own implementation.

That being said, if you're wanting to do advanced regex have you considered just iterating the arrays manually and applying the regex?
